configure.ac

```diff
@@ -2502,14 +2502,14 @@ if test "x$enable_python" = "xyes" || \
     AC_MSG_ERROR([only specify one of PYTHON or PYTHON_VERSION])
   fi
 
-  if test -n "$PYTHON"; then
-    PYTHON_VERSION=`$PYTHON -c "import sys; sys.stdout.write(sys.version[[:3]])"`
-  fi
-
   if test -n "$PYTHON_VERSION"; then
     AC_SUBST([PYTHON], [python$PYTHON_VERSION])
   fi
 
+  if test -n "$PYTHON"; then
+    PYTHON_VERSION=`$PYTHON -c "import sys; sys.stdout.write(sys.version[[:3]])"`
+  fi
+
   if test -z "$PYTHON" && test -z "$PYTHON_VERSION"; then
     AM_PATH_PYTHON()
   fi
```

Comment:
What does this do? Shouldn't this come before we start setting PYTHON/PYTHON_VERSION in order to populate them when the user didn't provide either?

Reply:
Put this into an "else" clause. I.e. if neither PYTHON/PYTHON_VERSION is provided, then initialize both.